### ISSUE TITLE
AzureBlobFileSystem to support CertificateCredential

### ISSUE THREAD

**Issue by UnderTheCraft** [fsspec/adlfs#399]
Due to restriction of a security concept, I am not able to authenticate using a TokenCredential, as [specified for the AzureBlobFileSystem](https://github.com/fsspec/adlfs/blob/2d56e411906519f31ee614e601469f67845b41bd/adlfs/spec.py#L141-L143).
Instead, I can only work with the [CertificateCredential from azure.identity](https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.certificatecredential?view=azure-python).

Would it be possible to add a support for this?

**Comment by TomAugspurger** [comment]
Does a CertificateCredential work with azure.storage.blob.ContainerClient?
https://learn.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.containerclient?view=azure-python

If it works with that, then it should work through adlfs.

**Comment by UnderTheCraft** [comment]
> Does a CertificateCredential work with azure.storage.blob.ContainerClient?
No, as far as I can see, the ContainerClient can only be created using an ["AzureNamedKeyCredential", "AzureSasCredential" or "TokenCredential"](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py#L183).

I actually already found a solution to read/write partitioned parquet with a CertificateCredential, using [azure.storage.filedatalake.DataLakeServiceClient](https://learn.microsoft.com/en-us/python/api/azure-storage-file-datalake/azure.storage.filedatalake.datalakeserviceclient?view=azure-python) and another library called ['pyarrowfs_adlgen2' from Robin Kåveland](https://github.com/kaaveland/pyarrowfs-adlgen2).

What I think is interesting here, is that the [constructor of the DataLakeServiceClient class](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_service_client.py#L76-L78) also does not mention the usage of an CertificateCredential explicitly - just the three Credential options, I also mentioned above.
So the CertificateCredential might also work somehow with the adlfs lib, but since I have a working solution now, I probaly won't test this.

However, in case anyone else is facing the same issue, here the code I used:
```python
from azure.storage.filedatalake import DataLakeServiceClient
from azure.identity import CertificateCredential
import pyarrowfs_adlgen2
import pyarrow
import pandas

credential = CertificateCredential(tenant_id, client_id, certificate_data=certificate)
datalake_service_client= DataLakeServiceClient(account_url=account_url, credential=credential)

handler = pyarrowfs_adlgen2.AccountHandler(datalake_service=datalake_service_client)
filesystem = pyarrow.fs.PyFileSystem(handler)

df = pandas.read_parquet(path=path, filesystem=filesystem)
```

@TomAugspurger feel free to close this issue, if no further action required from your side.
